### PR TITLE
WIP: Task 88 use all units from repos loc

### LINKED ISSUE #88: Use all units from a repository location

### PR DESCRIPTION
**Pull request by Anna-Lena-DLR**
Added "addAll" to grammar when specifiying a unit. It adds all units of a repository location to the .target file.

grafik

closes #88

### CODE REVIEW

File: de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/contentassist/UnitManager.xtend [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import org.eclipse.core.runtime.Status
+import org.eclipse.core.runtime.CoreException
+import java.io.IOException
+import de.dlr.sc.overtarget.language.Activator
+import org.eclipse.ui.statushandlers.StatusManager
+import de.dlr.sc.overtarget.language.targetmodel.TargetFile
+
+class UnitManager {
+	
+	static HashMap<String, ArrayList<Unit>> mapOfUnits = new HashMap<String, ArrayList<Unit>>
```
Comment (franzTobiasDLR) [inline de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/contentassist/UnitManager.xtend:31]: I think having this class as [Singelton](https://de.wikibooks.org/wiki/Muster:_Java:_Singleton) would be cleaner than a static map here?

File: de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/contentassist/UnitManager.xtend [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	
+	def getMapOfUnits(){
+		return mapOfUnits
+	}
+	
+	def getUnitsOfReposLoc(TargetModel model) {
```
Comment (franzTobiasDLR) [inline de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/contentassist/UnitManager.xtend:37]: should this method have the RepoLoc as parameter? (As the method name implies) And then return a list of units (As the method name implies :D )

File: de.dlr.sc.overtarget.language/model/generated/Overtarget.genmodel [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
+    xmlns:genmodel="http://www.eclipse.org/emf/2002/GenModel" copyrightText="Copyright (c) 2018-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.&#xA;&#xA;This program and the accompanying materials are made available under the&#xA;terms of the Eclipse Public License 2.0 which is available at&#xA;http://www.eclipse.org/legal/epl-2.0.&#xA;&#xA;SPDX-License-Identifier: EPL-2.0"
     modelDirectory="/de.dlr.sc.overtarget.language/src-gen" modelPluginID="de.dlr.sc.overtarget.language"
     forceOverwrite="true" modelName="Overtarget" updateClasspath="false" rootExtendsClass="org.eclipse.emf.ecore.impl.MinimalEObjectImpl$Container"
-    complianceLevel="6.0" copyrightFields="false" runtimeVersion="2.18">
+    complianceLevel="8.0" copyrightFields="false" runtimeVersion="2.20">
```
Comment (franzTobiasDLR) [inline de.dlr.sc.overtarget.language/model/generated/Overtarget.genmodel:6]: Did you change that on purpose?

File: de.dlr.sc.overtarget.language/src/de/dlr/sc/overtarget/language/TargetModel.xtext [hunk 4]
```diff
@@ -60,7 +60,7 @@ SimpleVersion:
 ;
 
 Unit:
-	'Unit' source=Source 'version' vers= (Version | 'newest') ';'
+	{Unit} 'Unit' addAll?='addAll'? (source=Source 'version' vers= (Version | 'newest'))? ';'
```
Comment (franzTobiasDLR) [inline de.dlr.sc.overtarget.language/src/de/dlr/sc/overtarget/language/TargetModel.xtext:63]: Wouln't it make more senese to have this addAll option as paort of the RepoLoc? rathern than the Unit? I mean technically a Unit cannot be all Units, right? Not sure, just asking...

**Review by franzTobiasDLR — COMMENTED** [review]
Is it intent that this unit manager is already part of this PR?

File: de.dlr.sc.overtarget.language/src/de/dlr/sc/overtarget/language/TargetModel.xtext (hunk 4 above)
Comment (Anna-Lena-DLR) [reply de.dlr.sc.overtarget.language/src/de/dlr/sc/overtarget/language/TargetModel.xtext:63]: Yes, I think you're right. Thank you :)

File: de.dlr.sc.overtarget.language/model/generated/Overtarget.genmodel (hunk 3 above)
Comment (Anna-Lena-DLR) [reply de.dlr.sc.overtarget.language/model/generated/Overtarget.genmodel:6]: No, it was changed when I generated the xtext language.

**Comment by Anna-Lena-DLR** [comment]
new opened with #130

**Pull request closed without merging**